src/components/course/data/utils.jsx

```diff
@@ -135,18 +135,26 @@ export function findOfferForCourse(offers, catalogList) {
 
 const getBestCourseMode = (courseModes) => {
   const {
-    VERIFIED, PROFESSIONAL, NO_ID_PROFESSIONAL, AUDIT,
+    VERIFIED, PROFESSIONAL, NO_ID_PROFESSIONAL, AUDIT, HONOR,
   } = COURSE_MODES_MAP;
   /** Returns the 'highest' course mode available.
-    *  Modes are ranked ['verified', 'professional', 'no-id-professional', 'audit'] */
+    *  Modes are ranked ['verified', 'professional', 'no-id-professional', 'audit', 'honor'] */
   if (courseModes.includes(VERIFIED)) {
     return VERIFIED;
-  } if (courseModes.includes(PROFESSIONAL)) {
+  }
+  if (courseModes.includes(PROFESSIONAL)) {
     return PROFESSIONAL;
-  } if (courseModes.includes(NO_ID_PROFESSIONAL)) {
+  }
+  if (courseModes.includes(NO_ID_PROFESSIONAL)) {
     return NO_ID_PROFESSIONAL;
   }
-  return AUDIT;
+  if (courseModes.includes(AUDIT)) {
+    return AUDIT;
+  }
+  if (courseModes.includes(HONOR)) {
+    return HONOR;
+  }
```

Comment:
Would we not want `HONOR` above `AUDIT`?

Reply:
I'm going with the same order as defined in edx-enterprise, which places `HONOR` after `AUDIT`: https://github.com/edx/edx-enterprise/blob/f9e75ce97229611a4aa0ecef185588de20cecfed/enterprise/constants.py#L64

```python
# Course mode sorting based on slug
COURSE_MODE_SORT_ORDER = [
    CourseModes.VERIFIED,
    CourseModes.PROFESSIONAL,
    CourseModes.NO_ID_PROFESSIONAL,
    CourseModes.AUDIT,
    CourseModes.HONOR,
]
```